### PR TITLE
LPD-34076

### PR DESCRIPTION
**Pull request by gabrielwas**
https://liferay.atlassian.net/browse/LPD-34076

### CODE REVIEW

**Comment by gabrielwas** [comment]
ci:test:sf

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 1]
```diff
@@ -198,10 +179,14 @@ public static void setUpClass() throws Exception {
...
+		_pushServiceContext();
 	}
 
 	@AfterClass
 	public static void tearDownClass() throws Exception {
+		ServiceContextThreadLocal.popServiceContext();
+
 		if (_serviceRegistration != null) {
 			_serviceRegistration.unregister();
 		}
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:192]: I would follow the same order from the `setUpClass` method:

```
_serviceRegistration = bundleContext.registerService...

_pushServiceContext();
```

in the `tearDownClass` method:

```
if (_serviceRegistration != null) {
   _serviceRegistration.unregister();
}

ServiceContextThreadLocal.popServiceContext();
```

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 2]
```diff
@@ -1385,16 +1358,43 @@ private void _assertNotificationQueueEntryTermValues(
...
+		DateFormat dateFormat = DateFormatFactoryUtil.getSimpleDateFormat(
+			DateTimeFormatterBuilder.getLocalizedDateTimePattern(
+				FormatStyle.SHORT, FormatStyle.SHORT, IsoChronology.INSTANCE,
+				LocaleUtil.US),
+			LocaleUtil.US);
+
+		return dateFormat.format(date);
+	}
+
+	private String _getCommerceOrderExpectedBody(CommerceOrder commerceOrder)
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:1381]: I would rename it to something a little more explicit like  

1. `_getCommerceOrderNotificationQueueEntryBody` and 
2. `_getObjectEntryNotificationQueueEntryBody` 

or inline

```
private void _assertNotificationQueueEntryBody(Object object)
	throws Exception {

	List<NotificationQueueEntry> notificationQueueEntries =
		notificationQueueEntryLocalService.getNotificationEntries(
			NotificationConstants.TYPE_EMAIL,
			NotificationQueueEntryConstants.STATUS_SENT);

	Assert.assertEquals(
		notificationQueueEntries.toString(), 1,
		notificationQueueEntries.size());

	String expectedBody = null;

	if (object instanceof CommerceOrder) {
		CommerceOrder commerceOrder = (CommerceOrder) object;

		AccountEntry accountEntry = commerceOrder.getAccountEntry();

		expectedBody =
			StringUtil.merge(
				Arrays.asList(
					String.valueOf(accountEntry.getAccountEntryId()),
					String.valueOf(commerceOrder.isInactive()),
					commerceOrder.getName(),
					_formatDate(commerceOrder.getOrderDate()),
					String.valueOf(commerceOrder.isPending()),
					_formatDate(commerceOrder.getRequestedDeliveryDate()),
					_formatDate(commerceOrder.getStatusDate())),
				StringPool.NEW_LINE);
	}
	else if (object instanceof ObjectEntry) {
		ObjectEntry objectEntry = (ObjectEntry) object;

		com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry =
			_objectEntryLocalService.getObjectEntry(objectEntry.getId());

		expectedBody = StringUtil.merge(
			Arrays.asList(
				LanguageUtil.getLanguageId(LocaleUtil.US),
				serviceBuilderObjectEntry.getUserName(),
				_formatDate(serviceBuilderObjectEntry.getCreateDate()),
				serviceBuilderObjectEntry.getExternalReferenceCode(),
				_formatDate(serviceBuilderObjectEntry.getModifiedDate()),
				String.valueOf(serviceBuilderObjectEntry.getObjectEntryId()),
				_formatDate(serviceBuilderObjectEntry.getLastPublishDate()),
				WorkflowConstants.getStatusLabel(
					serviceBuilderObjectEntry.getStatus()),
				"true", "9/25/24 12:00 AM", "2024-09-25T00:00",
				"test@liferay.com", "12345", "123456789",
				"listTypeEntry1Value,listTypeEntry2Value",
				"listTypeEntry1Value", "", "textObjectFieldValue",
				_portal.getPortalURL(
					ServiceContextThreadLocal.getServiceContext(
					).getRequest())),
			StringPool.NEW_LINE);
	}
	else {
		expectedBody = (String) object;
	}

	NotificationQueueEntry notificationQueueEntry =
		notificationQueueEntries.get(0);

	Assert.assertEquals(expectedBody, notificationQueueEntry.getBody());

	notificationQueueEntryLocalService.deleteNotificationQueueEntry(
		notificationQueueEntry);
}
```

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 1 above)
Comment (gabrielwas) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:192]: I followed the "stack" concept here. That way, the unregister process will follow the same context it had on the setUpClass.

**Comment by gabrielwas** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:report:27754908

**Comment by carolmariaabb** [comment]
Possible Unique failures:

1. [LocalFile.ObjectActions#CanEditExpression
](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794488)
2. [LocalFile.ObjectUpgrade#ViewActionWithNotificationAfterUpgrade7413U33
](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794352)

cc @victorKammerer

**Comment by gabrielwas** [comment]
@carolmariaabb I've run [LocalFile.ObjectActions#CanEditExpression](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794488) and it passed locally. 
And for [LocalFile.ObjectUpgrade#ViewActionWithNotificationAfterUpgrade7413U33](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794352), it's a flaky one (we have some other similar failures in the history.

Thanks for the analysis!

**Comment by gabrielwas** [comment]
@carolmariaabb I've run [LocalFile.ObjectActions#CanEditExpression](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794488) and it passed locally. 
And for [LocalFile.ObjectUpgrade#ViewActionWithNotificationAfterUpgrade7413U33](https://testray.liferay.com/#/project/35392/routines/45357/build/27754908/case-result/27794352), it's a flaky one (we have some other similar failures in the history.

Thanks for the analysis!

**Comment by gabrielwas** [comment]
ci:forward

**Comment by gabrielwas** [comment]
ci:forward

**Comment by brittneyq** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
due to CI outage, sent to brian here: https://github.com/brianchandotcom/liferay-portal/pull/154734

**Pull request closed without merging**